### DISCUSSION TITLE
New blog post: The quest for Electron LTS

### DISCUSSION THREAD

**Discussion by Daeraxa** [pulsar-edit/.github#250] [Announcements]
In the beginning, Atom appeared. It created an API to make packages, but together with this API, it also allowed authors to use web APIs together with node.js packages, modules (including "native modules" - more on that later) and, finally, a special API that was used to communicate between the "main module" and the "browser part".

That last part, eventually, split from Atom and became Electron. And for a while, the Atom development was tied to the Electron one, meaning that an update on Atom usually meant an update on Electron, and vice-versa.

Unfortunately, that wasn't the case for a long time...

Check out the full post on the [Pulsar Blog!](https://pulsar-edit.dev/blog/20240124-mauricioszabo-the-quest-for-electron-lts.html)

**Comment by lifeisbard** [comment]
Thank you. Keep up the good work!

**Comment by Charadon** [comment]
Doesn't this mean that wayland support is now just over the horizon?

Also, which branch is all this work on? I'd love to test it out =D

**Reply by confused-Techie** [reply]
There's fingers crossed about a whole lot just over the horizon, but I'll have to double check Wayland support.

But if you'd like to try it out, the branch the most up to date I believe is [`feature/latest-electron-native-superstring`](https://github.com/pulsar-edit/pulsar/pull/713).

There's a few other branches that started this effort, but this looks to be the one with all of the changes. But @mauricioszabo any suggestions for anyone wanting to test drive this before we start packing bins for it?

**Reply by Daeraxa** [reply]
I've been using Pulsar on Wayland for a while. Or is that using xwayland or something, not an area I've really looked into.

**Reply by mauricioszabo** [reply]
Well, I re-enabled the Cirrus builds on my own fork - https://cirrus-ci.com/github/mauricioszabo/pulsar/dreamy-pulsar - because I don't use nearly enough all credits.

Granted, this branch have a little more stuff merged in, but it's latest electron nonetheless :)

**Reply by Charadon** [reply]
Probably xwayland if you haven't explicity passed some flags to enable it. You can find out by using `xwininfo -tree -root` and seeing if Pulsar is on the list.

**Reply by Charadon** [reply]
So, I got around to testing that branch, and launching it in Wayland mode. It works really well, with just a couple of issues:

1. The menubar doesn't show up at start, I have to go into settings to toggle it off and on. (Wouldn't even response to me pressing ALT, this happened under Xorg too.)
2. The autocompletion is a bit jank. Doesn't find as many results as it usually did, wouldn't show a autocomplete suggestion until I already typed out the whole function, and the Vala autocompletion just flat out wouldn't work.
3. The superstring dependency in packages.json needs to point to at least commit https://github.com/pulsar-edit/superstring/commit/1cb70e4f8ed4427f305c745c79079c98070f086a, as I couldn't build it due to missing the stdint stuff.

Overall, good stuff guys!

**Reply by mauricioszabo** [reply]
@Charadon thanks for testing! For Autocomplete, can you check if there's some error in devtools?

**Reply by Charadon** [reply]
![image](https://github.com/pulsar-edit/.github/assets/24775301/17a38746-ba73-442f-bc2b-9260c35689ef)
Here's a screenshot of what pops up when I type something.

**Reply by mauricioszabo** [reply]
Thanks a lot, @Charadon. That is a known bug - the version of Superstring that comes with the new Electron changed to a more stable API, and there are some undocumented methods and properties (and also implementation details) that are still in use by some packages. I remember fixing this error, but probably there are still other places that it uses. I'll handle this case, thanks again!

**Reply by Charadon** [reply]
Awesome! Also, I found out that if you set Auto Hide Menu Bar to enabled, the menu bar will response to Alt presses from each launch forward. However, if it's disabled, it won't respond anymore until you toggle it. If I had to take a guess, the menu bar is probably not set to be shown at start anymore, and since Auto Hide Menu Bar isn't on, it doesn't respond to Alt.

**Reply by mauricioszabo** [reply]
Weird because I am also using the newer Electron branch, and mine _do respond_ to Alt presses (but I'm on Xorg)

**Reply by Charadon** [reply]
Well, once you guys start releasing those binaries officially, i'll definitely help you guys test and iron out all those weird things =)

Also, I think all this would allow distro maintainers to make pulsar use a shared version of electron, so that means less memory usage!
